class.jetpack-settings-sync.php

```diff
@@ -0,0 +1,338 @@
+		/*
+		'sharing_button_style'         	 => 'string',
+		'sharing_label'                	 => 'string',
+		'sharing_show'                 	 => 'string',
+		'sharing_open_links'           	 => 'string',
+		*/
+	);
+
+	/**
+	 * Data that we want to sync.
+	 * @var array
+	 */
+	private static $mock_options = array(
+		'admin_url' => array(
+			'type' => 'url',
+			'callback' => 'get_admin_url'
+		),
+		'jetpack_protect_whitelist' => array(
+			'type' => array( 'local' => 'array', 'global' => 'array' ),
+			'callback' => 'jetpack_protect_format_whitelist'
+		),
+		'post_categories' => array(
+			'type' => 'array',
+			'callback' => array( 'Jetpack_Sync_Settings', 'get_post_caregories' )
+		),
+		'jetpack_sync_non_public_post_stati' => array(
+			'type' => 'bool',
+			'callback' => array( 'Jetpack_Options','get_option' ),
+			'callback_args' => array( 'sync_non_public_post_stati' )
+		),
+
+
+		'jetpack_relatedposts_allowed' => array(
+			'type' => 'bool',
+			'callback' => array( 'Jetpack_Sync_Settings','jetpack_relatedposts_supported' )
+		),
+
+		'jetpack_relatedposts_enabled' => array(
+			'type' => 'bool',
+			'callback' => array( 'Jetpack_Sync_Settings','jetpack_related_posts_data' ),
+			'callback_args' => array( 'enabled' )
+		),
+		'jetpack_relatedposts_show_headline' => array(
+			'type' => 'bool',
+			'callback' => array( 'Jetpack_Sync_Settings','jetpack_related_posts_data' ),
+			'callback_args' => array( 'show_headline' )
+		),
+		'jetpack_relatedposts_show_thumbnails' => array(
+			'type' =>  'bool',
+			'callback' => array( 'Jetpack_Sync_Settings','jetpack_related_posts_data' ),
+			'callback_args' => array( 'show_thumbnails' )
+		)
+
+	);
+
+	/**
+	 * Constants that we want to sync
+	 * @var array
+	 */
+	private static $constants = array(
+		'EMPTY_TRASH_DAYS' => 'int',
+		'WP_POST_REVISIONS'=> 'int',
+		'AUTOMATIC_UPDATER_DISABLED'=> 'bool',
+		'WP_AUTO_UPDATE_CORE'=> 'bool'
+	);
+	/**
+	 * This lets us add to the different to be synced.
+	 *
+	 * @param  $name [description]
+	 * @param [type] $type [description]
+	 * @param [type] $args [description]
+	 */
+	static function add_setting( $name, $type, $args ) {
+		switch( $type ) {
+			case 'option':
+				if ( ! array_key_exists( $name, self::$options ) ) {
+					self::$options[ $name ] = $args;
+				}
+				break;
+			case 'mock_option':
+				if ( ! array_key_exists( $name, self::$mock_options ) ) {
+					self::$mock_options[ $name ] = $args;
+				}
+				break;
+			case 'constant':
+				if ( ! array_key_exists( $name, self::$constants ) ) {
+					self::$constants[ $name ] = $args;
+				}
+				break;
+		}
+	}
+	/**
+	 * Set ndividual setting
+	 *
+	 * @param  sting $name
+	 * @param  string $type
+	 * @param  string or array $callback
+	 * @param  boolean $is_constant
+	 * @return value of the setting
+	 */
+	static function get( $name ) {
+
+		// Options
+		if ( is_array( self::$options ) &&  array_key_exists( $name, self::$options ) ) {
```

Comment:
I would remove the `is_array` check, as we are sure this is going to be an array.